drivers/timer/npcx_itim_timer.c

```diff
@@ -0,0 +1,250 @@
+			;
+	}
+
+	k_spinlock_key_t key = k_spin_lock(&lock);
+	uint32_t delta_ticks = (uint32_t)((npcx_itim_get_sys_cyc64() -
+				cyc_sys_announced) / SYS_CYCLES_PER_TICK);
+
+	/* Store announced cycles of system timer */
+	cyc_sys_announced = npcx_itim_get_sys_cyc64();
+	k_spin_unlock(&lock, key);
+
+	/* Informs the kernel that the specified number of ticks have elapsed */
+	z_clock_announce(IS_ENABLED(CONFIG_TICKLESS_KERNEL) ? delta_ticks : 1);
+}
+
+/* System timer api functions */
+void z_clock_set_timeout(int32_t ticks, bool idle)
+{
+	ARG_UNUSED(idle);
+
+	if (!IS_ENABLED(CONFIG_TICKLESS_KERNEL)) {
+		/* Only for tickless kernel system */
+		return;
+	}
+
+	LOG_DBG("timeout is %d", ticks);
+	/* Start a event timer in ticks */
+	npcx_itim_start_evt_tmr_by_tick(ticks);
+}
+
+uint32_t z_clock_elapsed(void)
+{
+	if (!IS_ENABLED(CONFIG_TICKLESS_KERNEL)) {
+		/* Always return 0 for tickful kernel system */
+		return 0;
+	}
+
+	k_spinlock_key_t key = k_spin_lock(&lock);
+	uint64_t current = npcx_itim_get_sys_cyc64();
+
+	k_spin_unlock(&lock, key);
+
+	/* Return how many ticks elapsed since last z_clock_announce() call */
+	return (uint32_t)((current - cyc_sys_announced) / SYS_CYCLES_PER_TICK);
+}
+
+uint32_t z_timer_cycle_get_32(void)
+{
+	k_spinlock_key_t key = k_spin_lock(&lock);
+	uint64_t current = npcx_itim_get_sys_cyc64();
+
+	k_spin_unlock(&lock, key);
+
+	/* Return how many cycles since system kernel timer start counting */
+	return (uint32_t)(current);
+}
+
+int z_clock_driver_init(const struct device *device)
+{
+	ARG_UNUSED(device);
+	int ret;
+	const struct device *const clk_dev =
+					device_get_binding(NPCX_CLK_CTRL_NAME);
+
+	/* Turn on all itim module clocks used for counting */
+	for (int i = 0; i < ARRAY_SIZE(itim_clk_cfg); i++) {
+		ret = clock_control_on(clk_dev, (clock_control_subsys_t *)
+				&itim_clk_cfg[i]);
+		if (ret < 0) {
+			LOG_ERR("Turn on timer %d clock failed.", i);
+			return ret;
+		}
+	}
+
+	/*
+	 * Step 1. Use a ITIM64 timer as system kernel timer for counting.
+	 * Configure 64-bit timer counter and its prescaler to 1 first.
+	 */
+	sys_tmr->ITPRE = 0;
+	sys_tmr->ITCNT64L = NPCX_ITIM64_MAX_HALF_CNT;
+	sys_tmr->ITCNT64H = NPCX_ITIM64_MAX_HALF_CNT;
+	/*
+	 * Select APB2 clock which freq is CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC,
+	 * and clear timeout status bit before enabling the whole module.
+	 */
+	sys_tmr->ITCTS = BIT(NPCX_ITCTS_TO_STS);
+	/* Enable 64-bit timer and start ticking */
+	sys_tmr->ITCTS |= BIT(NPCX_ITCTS_ITEN);
+
+	/*
+	 * Step 2. Use a ITIM32 timer for event handling (ex. timeout event).
+	 * Configure 32-bit timer's prescaler to 1 first.
+	 */
+	evt_tmr->ITPRE = 0;
+	/*
+	 * Select low frequency clock source (The freq is 32kHz), enable its
+	 * interrupt/wake-up sources, and clear timeout status bit before
+	 * enabling it.
+	 */
+	evt_tmr->ITCTS = BIT(NPCX_ITCTS_CKSEL) | BIT(NPCX_ITCTS_TO_WUE)
+			 | BIT(NPCX_ITCTS_TO_IE) | BIT(NPCX_ITCTS_TO_STS);
```

Comment:
The datasheet indicates to wait 92us after changing CLKSEL before enabling the module.  Probably a good idea to wait inline after updates to ITCTS.

Reply:
Indeed. I will add a delay later.

Reply:
Done.